### PR TITLE
Make `tabs(withType:)` private to prevent UI from relying on it

### LINKED ISSUE #352: Tabs button long press "Close All Tabs" should no longer show a number
Should show the number of tabs only for its specific mode (private vs regular) not the combined amount.

simulator screen shot - iphone x - 2018-10-23 at 16 39 50

### PR DESCRIPTION
**Pull request by kylehickinson**
Fixes #352 (Fixes the number, doesn't remove the number)

## Pull Request Checklist

- [ ] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`
- [x] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

_None included_

## Notes for testing this patch

Sanity check needed

### CODE REVIEW

**Comment by kylehickinson** [comment]
Rebased and fixed build error

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ class TabManager: NSObject {
         }
     }
 
-    fileprivate(set) var tabs = [Tab]()
+    fileprivate(set) var allTabs = [Tab]()
```
Comment (jhreis) [inline Client/Frontend/Browser/TabManager.swift:67]: This is kind of what I'm hoping to get away from. The only tabs that should be in-memory are the tabs for the current mode. If we go to private mode, the "tabs" array should be cleared out and filled with solely private tabs.

Going to normal mode, should dump all existing tabs, and switch to normal tabs. I don't think there should really be a concept of `allTabs`.

Comment (kylehickinson) [reply Client/Frontend/Browser/TabManager.swift:67]: Considering we no longer keep private tabs at all when the user leaves private mode, we could just re-query the tabs saved in CD when switching back to regular mode.

I think we should merge this anyways for the time being as a part 1 of many changes to TabManager.

We can make the next step removing `allTabs`/`tabsForCurrentMode` like this step was removing `tabs(withType:)`

Comment (jhreis) [reply Client/Frontend/Browser/TabManager.swift:67]: Sounds good, will merge, but we still need to refactor this API. Nice intermediate step though!

**Review by jhreis — APPROVED** [review]

**Pull request merged**